src/Symfony/Bundle/FrameworkBundle/Test/WebTestAssertions.php

```diff
@@ -0,0 +1,413 @@
+        );
+    }
+
+    public static function assertResponseHasCookie(string $name, string $path = '/', string $domain = null): void
+    {
+        $cookie = static::getResponseCookieFromClient($name, $path, $domain);
+
+        Assert::assertNotNull(
+            $cookie,
+            "Did not find expected cookie [$name]."
+        );
+    }
+
+    public static function assertResponseNotHasCookie(string $name, string $path = '/', string $domain = null): void
+    {
+        $cookie = static::getResponseCookieFromClient($name, $path, $domain);
+
+        Assert::assertNull(
+            $cookie,
+            "Cookie [$name] was not expected to be set."
+        );
+    }
+
+    public static function assertResponseCookieValueEquals(string $name, $expectedValue, string $path = '/', string $domain = null): void
+    {
+        $cookie = static::getResponseCookieFromClient($name, $path, $domain);
+
+        Assert::assertNotNull(
+            $cookie,
+            "Did not find expected cookie [$name]."
+        );
+        Assert::assertSame(
+            $expectedValue,
+            $value = $cookie->getValue(),
+            "Cookie name [$name] with value [$expectedValue] does not match actual value [$value]."
+        );
+    }
+
+    public static function assertResponseCookieValueNotEquals(string $name, $expectedValue, string $path = '/', string $domain = null): void
+    {
+        $cookie = static::getResponseCookieFromClient($name, $path, $domain);
+
+        Assert::assertNotNull(
+            $cookie,
+            "Did not find expected cookie [$name]."
+        );
+        Assert::assertNotSame(
+            $expectedValue,
+            $value = $cookie->getValue(),
+            "Cookie name [$name] with value [$expectedValue] was not expected to be equal to actual value [$value]."
+        );
+    }
+
+    public static function assertSelectorExists(string $selector): void
+    {
+        static::checkCrawlerAvailable();
+
+        $nodes = static::$client->getCrawler()->filter($selector);
+
+        Assert::assertGreaterThan(0, $nodes->count(), "Selector [$selector] does not resolve to any node.");
+    }
+
+    public static function assertSelectorNotExists(string $selector): void
+    {
+        static::checkCrawlerAvailable();
+
+        $nodes = static::$client->getCrawler()->filter($selector);
+
+        Assert::assertEquals(0, $count = $nodes->count(), "Selector [$selector] resolves to [$count] nodes where it expected 0.");
+    }
+
+    public static function assertSelectorContainsText(string $selector, string $text): void
+    {
+        static::checkCrawlerAvailable();
+
+        $nodes = static::$client->getCrawler()->filter($selector);
+
+        Assert::assertGreaterThan(0, $nodes->count(), "Selector [$selector] does not resolve to any node.");
+
+        Assert::assertContains($text, $nodes->text(), "Selector [$selector] does not contain text [$text].");
```

Comment:
Does it check against the first node of the list intentionally?
If not, is there a way to assert against all nodes?

same for `assertSelectorNotContainsText`

Reply:
Yep, I think checking _all_ nodes might be a bit cumbersome, and it probably corresponds to non-common behavior